codehearts/remembear
# Add reminders

## Description of changes

- Provides a reminder data interface to add, update, remove, and get by ID
- Provides a CLI interface to manage reminders in the database
- Provides some mocked dependencies which can be used in unit testing

## Reason for changes 

- To support reminder management
- To pave the way towards sorting reminders and executing them on a schedule

## Test steps 

1. Run automated tests

        cargo test
1. Add a new reminder to the database, you will see uid 1 and the given name/schedule output

        cargo run reminder add 2:53 '{"wed":["14:53:00"]}' 1 2
1. List all reminders, you should see the same reminder from the last command

        cargo run reminder list
1. Update the name and schedule for the reminder, you should see uid 1 and the updated name/schedule output

        cargo run reminder update 1 --name=2:54 -s '{"wed":["14:54:00"]}'
1. List all reminders, you should see the same reminder from the last command

        cargo run reminder list
1. Remove the reminder, you should see uid 1 and the name/schedule output

        cargo run reminder remove 1
1. List all reminders to confirm the database is empty

        cargo run reminder list
